src/saturn/stores/queues_store.py

```diff
@@ -0,0 +1,72 @@
+import datetime
+from typing import Union
+
+from sqlalchemy import or_
+from sqlalchemy import select
+from sqlalchemy.orm import joinedload
+
+from saturn.database import AnyAsyncSession
+from saturn.database import AnySession
+from saturn.models import Queue
+
+
+def create_queue(
+    *,
+    session: Union[AnySession],
+    pipeline: str,
+) -> Queue:
+    queue = Queue(pipeline=pipeline)
+    session.add(queue)
+    return queue
+
+
+async def get_assigned_queues(
+    *,
+    session: AnyAsyncSession,
+    worker_id: str,
+    assigned_after: datetime.datetime,
+) -> list[Queue]:
+    assigned_jobs: list[Queue] = (
+        (
+            await session.execute(
+                select(Queue)
+                .options(
+                    joinedload(Queue.job),
+                )
+                .where(Queue.assigned_to == worker_id)
+                .where(Queue.assigned_at >= assigned_after)
+                .order_by(Queue.id)
+            )
+        )
+        .scalars()
+        .all()
+    )
+    return assigned_jobs
+
+
+async def get_unassigned_queues(
+    *,
+    session: AnyAsyncSession,
+    assigned_before: datetime.datetime,
+    limit: int,
+) -> list[Queue]:
+    unassigned_queues: list[Queue] = (
+        (
+            await session.execute(
```

Comment:
We should have a [`FOR UPDATE`](https://www.postgresql.org/docs/current/sql-select.html#SQL-FOR-UPDATE-SHARE) somewhere to lock these items (And could do the same for above) and avoid commit error on every concurent call to `/lock`.

Reply:
When we discussed this in person we said we didn't want several workers doing assignations, so a mutex would be enough. This is already in the TODO list that I wrote in the view (see `lock.py`)